libp2p/protocols/kademlia/provider.nim

```diff
@@ -39,20 +80,48 @@ proc addProvider*(kad: KadDHT, cid: Cid) {.async: (raises: [CancelledError]), gc
       # Dispatch will timeout if any of the calls don't receive a response (which is normal)
       discard
 
+proc startProviding*(kad: KadDHT, c: Cid) {.async: (raises: [CancelledError]).} =
+  kad.providerManager.providedKeys.incl(c)
+  await kad.addProvider(c)
+
+proc stopProviding*(kad: KadDHT, c: Cid) =
+  kad.providerManager.providedKeys.excl(c)
```

Comment:
Should this also remove the key from `knownKeys` and the `records` that know about this cid?